### PR TITLE
CERT-22 - Add CSR feature

### LINKED ISSUE #22: Feature: Add CSR feature
### Is your feature request related to a problem? Please describe.

For now while issuing certs there is no feature for the client to request for certs which is a standard.

### Describe the feature you are requesting, as well as the possible use case(s) for it.

While issuing certs the client can provide a csr file which has the request for the certs.

### Indicate the importance of this feature to you.

Must-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by nyagamunene**
# What type of PR is this?

## Which issue(s) does this PR fix/relate to?

- Related Issue #22
- Resolves #22 

## Have you included tests for your changes?
Yes.

## Did you document any new/modified features?
Yes

### CODE REVIEW

**Comment by dborovcanin** [comment]
@nyagamunene What's the status with this one?

**Comment by nyagamunene** [comment]
> @nyagamunene What's the status with this one?

There is still some issues with tests which am currently addressing.

File: api/metrics.go [hunk 1]
```diff
@@ -135,3 +136,35 @@ func (mm *metricsMiddleware) GetChainCA(ctx context.Context, token string) (cert
...
+	defer func(begin time.Time) {
+		mm.counter.With("method", "create_csr").Add(1)
+		mm.latency.With("method", "create_csr").Observe(time.Since(begin).Seconds())
+	}(time.Now())
+	return mm.svc.CreateCSR(ctx, meta, entityID, key...)
+}
+
+func (mm *metricsMiddleware) SignCSR(ctx context.Context, csrID string, approve bool) error {
+	defer func(begin time.Time) {
+		mm.counter.With("method", "process_csr").Add(1)
```
Comment (dborovcanin) [inline api/metrics.go:150]:
```go
mm.counter.With("method", "sign_csr").Add(1)
```

File: postgres/csr/init.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	migrate "github.com/rubenv/sql-migrate"
+)
+
+func Migration() *migrate.MemoryMigrationSource {
+	return &migrate.MemoryMigrationSource{
+		Migrations: []*migrate.Migration{
+			{
+				Id: "csr_1",
+				Up: []string{
+					`CREATE TABLE IF NOT EXISTS csr (
```
Comment (dborovcanin) [inline postgres/csr/init.go:17]: Double-check if this is aligned properly.

File: sdk/sdk.go [hunk 3]
```diff
@@ -232,6 +275,33 @@ type SDK interface {
...
+	//
+	//	response, _ := sdk.ListCSRs(sdk.PageMetadata{EntityID: "entity_id", Status: "pending"})
+	//	fmt.Println(response)
+	ListCSRs(pm PageMetadata) (CSRPage, errors.SDKError)
+
+	// RetrieveCSR retrieves a specific CSR by ID
+	//
+	//	response, _ := sdk.RetrieveCSR("csr_id")
+	//	fmt.Println(response)
+	RetrieveCSR(csrID string) (CSR, errors.SDKError)
```
Comment (dborovcanin) [inline sdk/sdk.go:304]: Same order remark.

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**